recipes/stlab/all/conanfile.py

```diff
@@ -14,45 +14,191 @@ class Stlab(ConanFile):
+        self.options["boost"].without_type_erasure = True
+        self.options["boost"].without_wave = True
+
+    def _fix_boost_components(self):
+        if self.settings.os != "Macos": return
+        if self.settings.compiler != "apple-clang": return
+        if Version(self.settings.compiler.version) >= "12": return
+
+        #
+        # On Apple we have to force the usage of boost.variant, because Apple's implementation of C++17 is not complete.
+        #
+        self.output.info("Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead.")
+        self.options.boost_optional = True
+        self.options.boost_variant = True
+
+    def _default_task_system(self):
+        if self.settings.os == "Macos": 
+            return "libdispatch"
+
+        if self.settings.os == "Windows":
+            return "windows"
+
+        if self.settings.os == "Emscripten":
+            return "emscripten"
+
+        return "portable"
+
+    def _configure_task_system_auto(self):
+        self.options.task_system = self._default_task_system()
+
+    def _configure_task_system_libdispatch(self):
+        if self.settings.os == "Linux":
+            if self.settings.compiler != "clang":
+                self.options.task_system = self._default_task_system()
+                self.output.info("Libdispatch requires Clang compiler on Linux. The task system is changed to {}.".format(self.options.task_system))
+                return
+
+        elif self.settings.os != "Macos":
+            self.options.task_system = self._default_task_system()
+            self.output.info("Libdispatch is not supported on {}. The task system is changed to {}.".format(self.settings.os, self.options.task_system))
+            return
+
+    def _configure_task_system_windows(self):
+        if self.settings.os != "Windows":
+            self.options.task_system = self._default_task_system()
+            self.output.info("Libdispatch is not supported on {}. The task system is changed to {}.".format(self.settings.os, self.options.task_system))
+            return
+
+    def _configure_task_system_emscripten(self):
+        if self.settings.os != "Emscripten":
+            self.options.task_system = self._default_task_system()
+            self.output.info("Libdispatch is not supported on {}. The task system is changed to {}.".format(self.settings.os, self.options.task_system))
+            return
+
+    def _configure_task_system(self):
+        if self.options.task_system == "auto":
+            self._configure_task_system_auto()
+        elif self.options.task_system == "libdispatch":
+            self._configure_task_system_libdispatch()
+        elif self.options.task_system == "windows":
+            self._configure_task_system_windows()
+        elif self.options.task_system == "emscripten":
+            self._configure_task_system_emscripten()
+
     def configure(self):
```

Comment:
```suggestion
    def validate(self):
```

https://blog.conan.io/2020/12/14/New-conan-release-1-32.html